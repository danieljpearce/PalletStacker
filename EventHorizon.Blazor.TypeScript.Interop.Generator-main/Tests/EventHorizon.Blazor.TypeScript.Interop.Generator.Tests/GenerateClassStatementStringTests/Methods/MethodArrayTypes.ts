declare module Static.Primitive {
    export class ExampleClass {
        arrayExample(): Array<string>;
        arrayLikeExample(): ArrayLike<string>;
        floatArrayExample(): FloatArray;
        float32ArrayExample(): Float32Array;
        indicesArrayExample(): IndicesArray;
        uint8ArrayExample(): Uint8Array;

        nullableArrayExample(): Nullable<Array<string>>;
        nullableArrayLikeExample(): Nullable<ArrayLike<string>>;
        nullableFloatArrayExample(): Nullable<FloatArray>;
        nullableFloat32ArrayExample(): Nullable<Float32Array>;
        nullableIndicesArrayExample(): Nullable<IndicesArray>;
        nullableUint8ArrayExample(): Nullable<Uint8Array>;
        nullableDeepImmutableArray(): DeepImmutable<Float32Array>;
    }
}
