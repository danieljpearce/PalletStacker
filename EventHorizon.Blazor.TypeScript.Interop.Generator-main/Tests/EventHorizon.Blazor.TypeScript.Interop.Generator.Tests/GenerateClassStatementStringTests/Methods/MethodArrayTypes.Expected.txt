/// Generated - Do Not Edit
namespace Static.Primitive
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods
        public string[] arrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<string>(
                new object[]
                {
                    new string[] { this.___guid, "arrayExample" }
                }
            );
        }

        public string[] arrayLikeExample()
        {
            return EventHorizonBlazorInterop.FuncArray<string>(
                new object[]
                {
                    new string[] { this.___guid, "arrayLikeExample" }
                }
            );
        }

        public decimal[] floatArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "floatArrayExample" }
                }
            );
        }

        public decimal[] float32ArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "float32ArrayExample" }
                }
            );
        }

        public decimal[] indicesArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "indicesArrayExample" }
                }
            );
        }

        public decimal[] uint8ArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "uint8ArrayExample" }
                }
            );
        }

        public string[] nullableArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<string>(
                new object[]
                {
                    new string[] { this.___guid, "nullableArrayExample" }
                }
            );
        }

        public string[] nullableArrayLikeExample()
        {
            return EventHorizonBlazorInterop.FuncArray<string>(
                new object[]
                {
                    new string[] { this.___guid, "nullableArrayLikeExample" }
                }
            );
        }

        public decimal[] nullableFloatArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "nullableFloatArrayExample" }
                }
            );
        }

        public decimal[] nullableFloat32ArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "nullableFloat32ArrayExample" }
                }
            );
        }

        public decimal[] nullableIndicesArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "nullableIndicesArrayExample" }
                }
            );
        }

        public decimal[] nullableUint8ArrayExample()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "nullableUint8ArrayExample" }
                }
            );
        }

        public decimal[] nullableDeepImmutableArray()
        {
            return EventHorizonBlazorInterop.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "nullableDeepImmutableArray" }
                }
            );
        }
        #endregion
    }
}